src/core/server/status/status_service.ts

```diff
@@ -44,13 +44,16 @@ export class StatusService implements CoreService<InternalStatusServiceSetup> {
     this.logger = coreContext.logger.get('status');
   }
 
-  public setup({ coreStatuses }: SetupDeps) {
-    const core$ = this.setupCoreStatus(coreStatuses);
+  public setup(core: SetupDeps) {
+    const core$ = this.setupCoreStatus(core);
     const overall$: Observable<ServiceStatus> = core$.pipe(
       map(coreStatus => {
-        this.logger.debug('Recalculating overall status');
-        return getSummaryStatus(coreStatus);
-      })
+        const summary = getSummaryStatus(coreStatus as any);
+        this.logger.debug(`Recalculated overall status`, { status: summary });
+        return summary;
+      }),
+      distinctUntilChanged(isDeepStrictEqual),
+      shareReplay(1)
```

Comment:
We do another distinctUntilChanged and shareReplay in `setupCoreStatus` so we can remove this one.

Reply:
Maybe the `shareReplay` could be removed but I think we could still produce duplicate statuses since the summary rollup may or may not be the same value.